content/ghc-9.12-cabal-3.14/index.md

```diff
@@ -0,0 +1,126 @@
++++
+title = "GHC 9.12 & cabal 3.14 releases"
+date = 2024-12-22
+[taxonomies]
+authors = ["Hécate"]
+categories = ["Cabal", "GHC"]
+tags = ["Release"]
++++
+
+To conclude the year 2024, the GHC and Cabal teams are happy to announce the release of GHC 9.12 and cabal 3.14.
+
+<!-- more -->
+
+
+Here are some highlights:
+
+## GHC 9.12
+
+### Language Extensions
+
+This release brings its lot of new and exciting extensions:
+* [Multiline string literals][MultilineStrings] to spread a piece of text over several lines;
+* [Or-Patterns][OrPatterns], allowing you to match on several patterns in a single `case` branch:
+
+```haskell
+data Sweet = Cupcake | Liquorice | Cookie | Raisins
+
+-- Without Or-Patterns
+tasty Cupcake = True
+tasty Cookie = True
+tasty _ = False
+
+-- With Or-Patterns
+tasty (Cupcake; Cookie) = True
+tasty (Liquorice; Raisins) = False
+```
+
+* [NamedDefaults][NamedDefaults] allows you to create type defaulting declarations for literals other than for the `Num` class. This mechanism is used to make sure that in the expression `print (6 + 7)`, the multiplication gets a concrete type like `Integer`, `Int` or `Double`, instead of the vague  `Num a => a`.  
+You can now apply this mechanism to other typeclasses, like:
+
+```haskell
+default IsString (Text, String)
+```
+
+Which means that in a module with [`OverloadedStrings`][OverloadedStrings], string literals `"like this"` will default to [`Text`][Text] instead of a vague polymorphic type.
+
+### Base Library
+
+* Improvements to backtraces and exceptions. Lots of them.
+
+#### ⚠️ Deprecations
+
+* The deprecation cycle of [GHC.Pack][GHC.Pack] has come to an end. Goodbye!
+* [GHC.Desugar][GHC.Desugar] will be removed in GHC 9.14.
+
+### Code Generation
+
+* Experimental support for the [RISC-V Platform](https://gitlab.haskell.org/ghc/ghc/-/issues/16783);
+* SIMD! In the x86: Most floating-point operations and some integer vector operations are supported 128-bit vectors **without LLVM**. [Get in touch](https://matrix.to/#/#ghc:matrix.org) to help with this effort
+* You can try the new experimental [`-fobject-determinism`][-fobject-determinism] flag to enable deterministic object code generation.
+
+#### ⚠️ Deprecations
+
+* Support 32-bit Windows & macOS/iOS has been dropped;
```

Comment:
```suggestion
* Support for 32-bit Windows and macOS/iOS has been dropped;
```

Reply:
Is it "MacOS" (with capital M)?

Reply:
The official spelling has changed in 2016 to the lower case, e. g., ["macOS Sequoia"](https://www.apple.com/uk/macos/macos-sequoia/).

Reply:
Thanks!